documentation/docs/14-types.md

```diff
@@ -3,3 +3,70 @@ title: Types
 ---
 
 **TYPES**
+
+### Generated types
+
+The [`RequestHandler`](#sveltejs-kit-requesthandler) and [`Load`](#sveltejs-kit-load) types both accept a `Params` argument allowing you to type the `params` object:
+
+```diff
+/// file: src/routes/[foo]/[bar]/[baz].js
+-/** @type {import('@sveltejs').RequestHandler} */
++/** @type {import('@sveltejs').RequestHandler<{ foo: string; bar: string; baz: string }>} */
+export async function get({ params }) {
+	// ...
+}
+```
+
+Needless to say, this is cumbersome to write out, and less portable (if you were to rename the `[foo]` directory to `[qux]`, the type would no longer reflect reality).
+
+To solve this problem, SvelteKit generates `.d.ts` files for each of your endpoints and pages:
+
+```ts
+/// file: .svelte-kit/types/src/routes/[foo]/[bar]/[baz].d.ts
+/// link: false
+import type { RequestHandler as GenericRequestHandler, Load as GenericLoad } from '@sveltejs/kit';
+
+export type RequestHandler<Body = any> = GenericRequestHandler<
+	{ foo: string; bar: string; baz: string },
+	Body
+>;
+
+export type Load<Props = Record<string, any>> = GenericLoad<
+	{ foo: string; bar: string; baz: string },
+	Props
+>;
+```
+
+These files can be imported into your endpoints and pages as siblings, thanks to the [`rootDirs`](https://www.typescriptlang.org/tsconfig#rootDirs) option in your TypeScript configuration:
+
+```js
+/// file: src/routes/[foo]/[bar]/[baz].js
+// @filename: [baz].d.ts
+import type { RequestHandler as GenericRequestHandler, Load as GenericLoad } from '@sveltejs/kit';
+
+export type RequestHandler<Body = any> = GenericRequestHandler<
+	{ foo: string, bar: string, baz: string },
+	Body
+>;
+
+// @filename: index.js
+// @errors: 2355
+// ---cut---
+/** @type {import('./[baz]').RequestHandler} */
+export async function get({ params }) {
+	// ...
+}
+```
+
+```svelte
+<script context="module">
+	/** @type {import('./[baz]').Load} */
+	export async function load({ params, fetch, session, stuff }) {
+		// ...
+	}
+</script>
+```
+
+> For this to work, your own `tsconfig.json` or `jsconfig.json` should extend from the generated `.svelte-kit/tsconfig.json`:
```

Comment:
Oh, I guess we are mentioning .svelte-kit here. I dunno. It is officially part of the API now that we want people to extend the config in it. I'm not sure whether that changes my opinion about mentioning it up higher instead of calling it a hidden folder.

Reply:
I would also add a note that that folder is not available immediately, so you'll need to run a command first (we also need to restart the language-server for it to resolve the extended tsconfig again, adding it to this list https://github.com/sveltejs/language-tools/blob/master/packages/svelte-vscode/src/extension.ts#L144).
What if we add a postinstall hook to the `package.json` of the create-svelte templates to immediately generate the types?

Reply:
Do `postinstall` hooks run after you run `npm install` on the repo whose `package.json` they're in, or do they run when you've just installed a package whose `package.json` has one?

Reply:
Just checked, they do. Will add a `svelte-kit init` command and a `"postinstall": "svelte-kit init"` run script

Reply:
actually i'll do that in a separate PR, because there might be some bikeshedding involved